test/fixtures/server.js

```diff
@@ -103,8 +104,59 @@ function createSSLServer (opts) {
         }
     }
 
-    server = https.createServer(options, function (req, resp) {
-        server.emit(req.url, req, resp);
+    server = https.createServer(options, function (req, res) {
+        server.emit(req.url, req, res);
+    });
+
+    enableServerDestroy(server);
+
+    return server;
+}
+
+/**
+ * Simple redirect server for tests that emit hit events on each request captured.
+ * Use the URL format: /<urlPath>/<numberOfRedirects>/<responseCode>
+ *
+ * @example
+ * var s = createRedirectServer();
+ * s.on('hit', function (req, res) {
+ *     console.log(req.location);
+ * });
+ * s.on('/foo', function (req, res)) {
+ *     // this is called when there is no redirect.
+ * }
+ * s.listen(3000, callback);
+ */
+function createRedirectServer () {
+    var server = http.createServer(function (req, res) {
+        var urlTokens,
+            numberOfRedirects,
+            responseCode,
+            redirectURL;
+
+        server.emit('hit', req, res);
+
+        // /<urlPath>/<numberOfRedirects>/<responseCode>
+        if ((/\/\d+\/\d{3}$/).test(req.url)) {
+            urlTokens = req.url.split('/');
+            numberOfRedirects = parseInt(urlTokens[urlTokens.length - 2], 10);
+            responseCode = parseInt(urlTokens[urlTokens.length - 1], 10);
+
+            // redirect until all hops are covered
+            if (numberOfRedirects > 1) {
+                redirectURL = urlTokens.slice(0, -2).join('/') + '/' + (numberOfRedirects - 1) + '/' + responseCode;
```

Comment:
```suggestion
                redirectURL = urlTokens.slice(0, -2).join('/') + `/${numberOfRedirects}/${responseCode}`;
```

Decrement `numberOfRedirects`  while initialization and check if `numberOfRedirects > 0`